From 9b611ad453be06cfb48d9352ef65166a432a8edd Mon Sep 17 00:00:00 2001
From: Zhipeng Xie <xiezhipeng1@huawei.com>
Date: Tue, 16 Nov 2021 20:21:31 +0800
Subject: [PATCH 22/37] kpatch-build: support CROSS_COMPILE

some Makefile use $(CROSS_COMPILE)gcc to compile .o
file, append CROSS_COMPILE to MAKEVARS to make it
run into kpatch-cc.

Signed-off-by: Zhipeng Xie <xiezhipeng1@huawei.com>
---
 kpatch-build/kpatch-build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kpatch-build/kpatch-build b/kpatch-build/kpatch-build
index a73bd1b..daa9a81 100755
--- a/kpatch-build/kpatch-build
+++ b/kpatch-build/kpatch-build
@@ -997,6 +997,7 @@ if [[ -n "$CONFIG_CC_IS_CLANG" ]]; then
 	MAKEVARS+=("HOSTCC=clang")
 else
 	MAKEVARS+=("CC=${KPATCH_CC_PREFIX}${GCC}")
+	MAKEVARS+=("CROSS_COMPILE=${KPATCH_CC_PREFIX}")
 fi
 
 if [[ -n "$CONFIG_LD_IS_LLD" ]]; then
-- 
2.33.0

